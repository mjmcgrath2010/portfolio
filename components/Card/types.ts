import { ReactElement } from "react";

interface CardProps {
  children: ReactElement | ReactElement[];
  className?: string;
}

export interface CardImageProps {
  src: string;
  title: string;
}

export default CardProps;
