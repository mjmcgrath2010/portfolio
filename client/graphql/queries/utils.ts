/*
 * Example usage:
 *   export const getStaticProps = () => mapQueriesToProps([{ posts: getPosts }]);
 */
type QueryProps = {
  props: Object;
};

export const mapQueriesToProps = async (queries: any) =>
  await (async () => {
    const data = await queries.reduce(
      async (acc: QueryProps, queryObj: Function) => {
        const [prop, fn] = Object.entries(queryObj)[0];
        try {
          const { props } = acc;
          const data = await fn();
          return {
            props: {
              ...props,
              [prop]: data,
            },
          };
        } catch (e) {
          return {
            props: {
              error: "Oops, something went wrong",
            },
          };
        }
      },
      { props: {} }
    );
    return data;
  })();
