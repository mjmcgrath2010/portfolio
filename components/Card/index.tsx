import React from "react";
import Typography from "../Typography";
import styled from "styled-components";

import CardProps, { CardImageProps } from "./types";

const Container = styled.div`
  box-shadow: ${({ theme }) => theme.shadows.main};
  height: 100%;
  max-width: 90%;
  min-height: 120px;
  padding: ${({ theme }) => theme.spacing.md};
  position: relative;
  margin: 0 auto;
  display: flex;
  flex-flow: column wrap;
  border-radius: ${({ theme }) => theme.borderRadius};
`;

const ImageContainer = styled.div`
  background: ${({ src }: { src: string }) =>
    `linear-gradient( rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3) ), url(${src}) center`};
  height: 50%;
  min-height: 150px;
  filter: grayscale(100%);
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: ${({ theme }) => theme.borderRadius};
  padding: ${({ theme }) => theme.spacing.md};
  text-align: center;

  &:hover {
    background: ${({ src }: { src: string }) =>
      `linear-gradient( rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.7) ), url(${src}) center`};
    cursor: pointer;
  }
`;

const CardImage = ({ src, title }: CardImageProps) => (
  <ImageContainer src={src}>
    <Typography fontStyle="body" color="light">
      {title}
    </Typography>
  </ImageContainer>
);

CardImage.defaultProps = {
  title: "Card Example",
  src: "https://via.placeholder.com/150",
};

const Card = ({ children, className }: CardProps) => (
  <Container className={className}>{children}</Container>
);

Card.Image = CardImage;

Card.defaultProps = {
  className: "",
};

export default Card;
