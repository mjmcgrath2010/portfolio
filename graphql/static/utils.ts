export const mapQueriesToProps = async (queries: any) =>
  await (async () => {
    const data = await queries.reduce(
      async (acc, queryObj) => {
        const [prop, fn] = Object.entries(queryObj)[0];
        try {
          const { props } = acc;
          const data = await fn();

          return {
            props: {
              ...props,
              [prop]: data,
            },
          };
        } catch (e) {
          return {
            error: e.message,
          };
        }
      },
      { props: {} }
    );
    return data;
  })();
