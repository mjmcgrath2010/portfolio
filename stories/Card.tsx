/* eslint-disable import/no-anonymous-default-export */
import React from "react";
import styled from "styled-components";
import Card from "../components/Card";

const Container = styled.div`
  max-width: 200px;
`;

export default {
  title: "Components/Card",
  component: Card,
  argTypes: {},
  parameters: {},
};

export const DefaultStory = () => (
  <Container>
    <Card>
      <Card.Image
        title="Example Featured Text"
        src="https://via.placeholder.com/150"
      />
    </Card>
  </Container>
);
DefaultStory.args = {};

DefaultStory.storyName = "Card - Default Story";
