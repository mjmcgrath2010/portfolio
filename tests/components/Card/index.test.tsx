import React from "react";
import Card from "@components/Card";

import { render, screen } from "@test/utils";

describe("<Card />", () => {
  it("matches the snapshot", () => {
    const view = render(<Card />);

    expect(view).toMatchSnapshot();
  });
});
