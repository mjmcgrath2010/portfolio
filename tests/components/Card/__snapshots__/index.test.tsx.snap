// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  box-shadow: hsl(0deg 0% 27% / 15%) 2px 3px 5px;
  height: 100%;
  max-width: 90%;
  min-height: 120px;
  padding: 16px;
  position: relative;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  border-radius: 5px;
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": .c0 {
  box-shadow: hsl(0deg 0% 27% / 15%) 2px 3px 5px;
  height: 100%;
  max-width: 90%;
  min-height: 120px;
  padding: 16px;
  position: relative;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  border-radius: 5px;
}

<div>
    <div
      class="c0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
